[OSDOCS#17050]: CQA 2.0 Multi-Architecture Compute Node Creation (#112721)
openshift/openshift-docs: modules/machine-user-infra-machines-pxe.adoc

TIMEOUT 20
PROMPT 0
LABEL pxeboot
    KERNEL http://<HTTP_server>/rhcos-<version>-live-kernel-<architecture>
    APPEND initrd=http://<HTTP_server>/rhcos-<version>-live-initramfs.<architecture>.img coreos.inst.install_dev=/dev/sda coreos.inst.ignition_url=http://<HTTP_server>/worker.ign coreos.live.rootfs_url=http://<HTTP_server>/rhcos-<version>-live-rootfs.<architecture>.img
----
+
where:
+
`KERNEL`:: Specifies the location of the live `kernel` file that you uploaded to your HTTP server.
`APPEND`:: Specifies the locations of the {op-system} files that you uploaded to your HTTP server:
`initrd`:: Specifies the location of the live `initramfs` file.
`coreos.inst.ignition_url`:: Specifies the location of the worker Ignition config file. This parameter only supports HTTP and HTTPS.

> `coreos.live.rootfs_url`:: Specifies the location of the live `rootfs` file. This parameter only supports HTTP and HTTPS.

Reviewer: nicely done, this is an example where making more items than were previously in the callout list is acceptable, and even preferable. 

One small thing I would nitpick here, and it's not even something you wrote yourself, but - when using the [adverb "only"](https://www.ibm.com/docs/en/ibm-style?topic=grammar-adverbs-only), we want to bring it as close to the phrase it modifies as we can for the sake of specificity. In this case I think we can shift it slightly:

```suggestion
`coreos.inst.ignition_url`:: Specifies the location of the worker Ignition config file. This parameter supports only HTTP and HTTPS.
`coreos.live.rootfs_url`:: Specifies the location of the live `rootfs` file. This parameter supports only HTTP and HTTPS.
```

+
[NOTE]
====

> This configuration does not enable serial console access on machines with a graphical console. To configure a different console, add one or more `console=` arguments to the `APPEND` line. For example, add `console=tty0 console=ttyS0` to set the first PC serial port as the primary console and the graphical console as a secondary console. For more information on setting up a serial terminal and/or console in {op-system}, see the Additional resources section.

Reviewer: This isn't super problematic but in general, it's best practice to still refer to the name of the link you're providing as an additional resource, even if it would be pretty self explanatory once a user looked at the additional resources section:

```suggestion
This configuration does not enable serial console access on machines with a graphical console. To configure a different console, add one or more `console=` arguments to the `APPEND` line. For example, add `console=tty0 console=ttyS0` to set the first PC serial port as the primary console and the graphical console as a secondary console. For more information on setting up a serial terminal and/or console in {op-system}, see "How does one set up a serial terminal and/or console in Red Hat Enterprise Linux?".
```

> This configuration does not enable serial console access on machines with a graphical console. To configure a different console, add one or more `console=` arguments to the `APPEND` line. For example, add `console=tty0 console=ttyS0` to set the first PC serial port as the primary console and the graphical console as a secondary console. For more information on How does one set up a serial terminal and/or console in Red Hat Enterprise Linux, see Additional resources section.

Reviewer 2: ```suggestion
This configuration does not enable serial console access on machines with a graphical console. To configure a different console, add one or more `console=` arguments to the `APPEND` line. For example, add `console=tty0 console=ttyS0` to set the first PC serial port as the primary console and the graphical console as a secondary console. For more information on setting up a serial terminal and/or console in {op-system}, see the Additional resources section.
```

endif::ibm-power[]
+
----
kernel http://<HTTP_server>/rhcos-<version>-live-kernel-<architecture> initrd=main coreos.live.rootfs_url=http://<HTTP_server>/rhcos-<version>-live-rootfs.<architecture>.img coreos.inst.install_dev=/dev/sda coreos.inst.ignition_url=http://<HTTP_server>/worker.ign
initrd --name main http://<HTTP_server>/rhcos-<version>-live-initramfs.<architecture>.img
boot
----
+
where:
+
`kernel`:: Specifies the location of the `kernel` file that you uploaded to your HTTP server.
`initrd=main`:: Specifies an argument that is required for booting on UEFI systems.
`coreos.live.rootfs_url`:: Specifies the location of the `rootfs` file that you uploaded to your HTTP server.
`coreos.inst.ignition_url`:: Specifies the location of the worker Ignition config file that you uploaded to your HTTP server.

> `ip`:: Specifies a single interface to use if you have multiple network interface cards (NICs). For example, to use DHCP on a NIC named `eno1`, set `ip=eno1:dhcp`.

Reviewer: This is a tough item to handle. It's part of an existing callout description, but there is no `ip` option shown in the example (or maybe it was removed). I don't think it would make sense to keep this in the description list in this case, and instead I would add the previous text to the NOTE below:

```suggestion
```

`initrd --name main`:: Specifies the location of the `initramfs` file that you uploaded to your HTTP server.
+
[NOTE]
====

> This configuration does not enable serial console access on machines with a graphical console To configure a different console, add one or more `console=` arguments to the `kernel` line. For example, add `console=tty0 console=ttyS0` to set the first PC serial port as the primary console and the graphical console as a secondary console. For more information on setting up a serial terminal and/or console in {op-system}, see the Additional resources section and "Enabling the serial console for PXE and ISO installation" in the "Advanced {op-system} installation configuration" section.

Reviewer: This is how the `ip` description would look like in the note:

```suggestion
* If you use multiple NICs, specify a single interface in the `ip` option.
For example, to use DHCP on a NIC that is named `eno1`, set `ip=eno1:dhcp`.

* This configuration does not enable serial console access on machines with a graphical console To configure a different console, add one or more `console=` arguments to the `kernel` line. For example, add `console=tty0 console=ttyS0` to set the first PC serial port as the primary console and the graphical console as a secondary console. For more information on setting up a serial terminal and/or console in {op-system}, see the Additional resources section and "Enabling the serial console for PXE and ISO installation" in the "Advanced {op-system} installation configuration" section.
```

> This configuration does not enable serial console access on machines with a graphical console To configure a different console, add one or more `console=` arguments to the `kernel` line. For example, add `console=tty0 console=ttyS0` to set the first PC serial port as the primary console and the graphical console as a secondary console. For more information on setting up a serial terminal and/or console in {op-system}, see the Additional resources section and "Enabling the serial console for PXE and ISO installation" in the "Advanced {op-system} installation configuration" section.

Reviewer: Also, same comment about using the actual name of the link you are providing below. In this case, since the sentence also mentions another file elsewhere, I would keep the mention of the additional resources section:

```suggestion
This configuration does not enable serial console access on machines with a graphical console To configure a different console, add one or more `console=` arguments to the `kernel` line. For example, add `console=tty0 console=ttyS0` to set the first PC serial port as the primary console and the graphical console as a secondary console. For more information on setting up a serial terminal and/or console in {op-system}, see "How does one set up a serial terminal and/or console in Red Hat Enterprise Linux?" in the Additional resources section and "Enabling the serial console for PXE and ISO installation" in the "Advanced {op-system} installation configuration" section.
```

> This configuration does not enable serial console access on machines with a graphical console To configure a different console, add one or more `console=` arguments to the `kernel` line. For example, add `console=tty0 console=ttyS0` to set the first PC serial port as the primary console and the graphical console as a secondary console. For more information on How does one set up a serial terminal and/or console in Red Hat Enterprise Linux, see Additional resources section and "Enabling the serial console for PXE and ISO installation" in the "Advanced {op-system} installation configuration" section.

Reviewer 2: ```suggestion
This configuration does not enable serial console access on machines with a graphical console To configure a different console, add one or more `console=` arguments to the `kernel` line. For example, add `console=tty0 console=ttyS0` to set the first PC serial port as the primary console and the graphical console as a secondary console. For more information on setting up a serial terminal and/or console in {op-system}, see the Additional resources section and "Enabling the serial console for PXE and ISO installation" in the "Advanced {op-system} installation configuration" section.
```

====
+
[NOTE]
====
ifndef::ibm-power[]

> To network boot the CoreOS `kernel` on `aarch64` architecture, you need to use a version of iPXE build with the `IMAGE_GZIP` option enabled. For more information, see the Additional resources section.

Reviewer: Same comment as before about keeping section names even when pointing users to the additional resources section. It's important that users know what to look for when you're providing additional information:
```suggestion
To network boot the CoreOS `kernel` on `aarch64` architecture, you need to use a version of iPXE build with the `IMAGE_GZIP` option enabled. For more information, see "`IMAGE_GZIP` option in iPXE".
```

Please also apply this feedback for any remaining instances of this kind of change you made in the PR

> To network boot the CoreOS `kernel` on `aarch64` architecture, you need to use a version of iPXE build with the `IMAGE_GZIP` option enabled. For more information, see Additional resources section.

Reviewer 2: ```suggestion
To network boot the CoreOS `kernel` on `aarch64` architecture, you need to use a version of iPXE build with the `IMAGE_GZIP` option enabled. For more information, see the Additional resources section.
```

endif::ibm-power[]
ifdef::ibm-power[]

> To network boot the CoreOS `kernel` on `ppc64le` architecture, you need to use a version of iPXE build with the `IMAGE_GZIP` option enabled. For more information, see Additional resources section.

Reviewer 2: ```suggestion
To network boot the CoreOS `kernel` on `ppc64le` architecture, you need to use a version of iPXE build with the `IMAGE_GZIP` option enabled. For more information, see the Additional resources section.
```

`coreos.inst.ignition_url`:: Specifies the location of the worker Ignition config file that you uploaded to your HTTP server.
`initrd --name main`:: Specifies the location of the `initramfs` file that you uploaded to your HTTP server.
+
[NOTE]
====
* If you use multiple NICs, specify a single interface in the `ip` option.
For example, to use DHCP on a NIC named `eno1`, set `ip=eno1:dhcp`.
+
* This configuration does not enable serial console access on machines with a graphical console To configure a different console, add one or more `console=` arguments to the `kernel` line. For example, add `console=tty0 console=ttyS0` to set the first PC serial port as the primary console and the graphical console as a secondary console. For more information on setting up a serial terminal and/or console in {op-system}, see "How does one set up a serial terminal and/or console in Red Hat Enterprise Linux?" in the Additional resources section and "Enabling the serial console for PXE and ISO installation" in the "Advanced {op-system} installation configuration" section.
====
+
[NOTE]
====
ifndef::ibm-power[]

> To network boot the CoreOS `kernel` on `aarch64` architecture, you need to use a version of iPXE build with the `IMAGE_GZIP` option enabled. For more information, see " `IMAGE_GZIP` option in iPXE".

Reviewer: Small nit, just an extra space here:

```suggestion
To network boot the CoreOS `kernel` on `aarch64` architecture, you need to use a version of iPXE build with the `IMAGE_GZIP` option enabled. For more information, see "`IMAGE_GZIP` option in iPXE".
```

+
----
menuentry 'Install CoreOS' {
    linux rhcos-<version>-live-kernel-<architecture>  coreos.live.rootfs_url=http://<HTTP_server>/rhcos-<version>-live-rootfs.<architecture>.img coreos.inst.install_dev=/dev/sda coreos.inst.ignition_url=http://<HTTP_server>/worker.ign
    initrd rhcos-<version>-live-initramfs.<architecture>.img
}
----
+
where:
+
`linux`:: Specifies the filename of the live `kernel` file on your TFTP server.
`coreos.live.rootfs_url`:: Specifies the HTTP or HTTPS URL of the live `rootfs` file.
`coreos.inst.ignition_url`:: Specifies the HTTP or HTTPS URL of the worker Ignition config file.

> `ip`:: Specifies a single network interface controller (NIC) to configure if you use multiple NICs. For example, to use DHCP on a NIC named `eno1`, set `ip=eno1:dhcp`.

Reviewer: As I mentioned before, because there's no `ip` anywhere in the example code block, I would take the previous wording and put it into a NOTE. In this case, there's no existing note, so create one and place it below this description list

> `initrd`:: Specifies the filename of the live `initramfs` file on your TFTP server.

Reviewer: All of these (except for `ip`) modifies the meaning of the description in a way that honestly kind of would require QE to validate, even if you have info from other content that substantiates your decision, like seeing that `coreos.live.rootfs_url` obviously defines a URL.

But honestly, for CQAs, if you don't _have to_ take the risk of modifying the meaning of something to just fix a vale error, I would personally just play it safe and don't risk changing the technical meaning of content. 
```suggestion
`linux`:: Specifies the location of the live `kernel` file on your TFTP server.
`coreos.live.rootfs_url`:: Specifies the location of the live `rootfs` file.
`coreos.inst.ignition_url`:: Specifies the location of the worker Ignition config file.
`ip`:: Specifies a single network interface controller (NIC) to configure if you use multiple NICs. For example, to use DHCP on a NIC named `eno1`, set `ip=eno1:dhcp`.
`initrd`:: Specifies the location of the live `initramfs` file on your TFTP server.
```

And in the case of using "filename", that's definitely too much of a risk, because IMO a filename and a location are not always the same (often a location will be a path to a file, not just the name of the file itself)
